### ISSUE TITLE
dataclass fields reported as having a default value when field metadata is assigned

### ISSUE THREAD

**Issue by roeap** [microsoft/pyright#798]
**Describe the bug**
When assigning meta data to a field in a dataclass, pyright recognizes it as having a default value and thus reports an error if a field with meta data is defined before a field without metadata. Normally we can easily work around this, but when inheriting from a data class with some meta data defined, I see no way to circumvent the message.

**To Reproduce**

```py
from dataclasses import dataclass, field

@dataclass
class SomeClass:
    prop1: str = field(metadata={"data": 0})
    prop2: List[str]
```
While there is no default value assigned to `prop1`, an error is reported for `prop2` being defined after a default value.

**Expected behavior**
Pyright should recongize a field as having a default value, if it is directly assigned a value, or the `field` function is called with either `default` of `default_factory` as one of the parameters.

**Screenshots or Code**
 see **To Reproduce**

**VS Code extension or command-line**
VS Code extension - 1.1.50

**Additional context**
n.a.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next published version of Pyright.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.51, which I just published. It will be included in the next release of Pylance.
